src/osd/PeeringState.cc

```diff
@@ -6698,7 +6698,9 @@ PeeringState::Deleting::Deleting(my_context ctx)
   : my_base(ctx),
     NamedState(context< PeeringMachine >().state_history, "Started/ToDelete/Deleting")
 {
+  start = ceph::mono_clock::now();
```

Comment:
`start` marks the time when the state machine transits to `Deleting`, and we calculate the duration from `start` to when the PG is deleted completely. but we already print log when the PG enters the `Deleting` state and when it exits this state. per https://www.boost.org/doc/libs/1_74_0/libs/statechart/doc/tutorial.html#TwoStageExit, `exit()` should be called when the state is destructed normally. i think that's the case we consider here, right? so we always have access to the timestamp, and hence the duration of the deletion? if you want to have a more visible log lines, probably, to note down the start time in `PeeringState::Deleting::Deleting()` and the end time in `PeeringState::Deleting::exit()` would suffice?

Reply:
I updated this part a bit and PeeringState::Deleting::exit does the print itself. Hence eliminating the need to pass 'start' out of this class.
But I'd like to preserve this specific logging (along with starting timestamp) for two reasons:
1) Printing duration on completion is more convenient for some purposes than having "start"/"terminate" line in a log. The removal task is pretty long (up to several days) and hence locating the starting event to learn the duration might be troublesome.
2) Followup PR is gonna add more OSD performance  counters and this requires starting timestamp for Deleting task.
See https://github.com/ceph/ceph/pull/37496/commits/ce14fcddd218ea3f29bfb799e5918b62665da0fa (this needs an update to reflect the latest changes in passing 'start' timestamp)